UI: Remove 'left' property from TooltipLinkList and Link components (#26324)
storybookjs/storybook: MIGRATION.md

Please file an issue if you need these APIs.

#### Removals in @storybook/components

> The `TooltipLinkList` component accepts a `links` property where, for each link, a `left` property could be passed. The left property is now removed in Storybook 8 and beyond. Use `icon` instead. The side-effect is that the `left` property is now removed from the `Link` component. The Link component is used to define `globalTypes` in the `preview.js` file, among other places:

Author: ```suggestion
The `TooltipLinkList` component accepts a `links` property where, for each link, a `left` property could be passed. The left property is now removed in Storybook 8 and beyond. Use `icon` instead. The side-effect is that the `left` property is now removed from the `Link` component. This has an effect on the `globalTypes` definition in the `preview.js` file, among other places:
```

> The `TooltipLinkList` component accepts a `links` property where, for each link, a `left` property could be passed. The left property is now removed in Storybook 8 and beyond. Use `icon` instead. The side-effect is that the `left` property is now removed from the `Link` component. This has an effect on the `globalTypes` definition in the `preview.js` file, among other places:
> 
> ```diff
> // Replace your-framework with the framework you are using (e.g., react, vue3)
> import { Preview } from '@storybook/your-framework';
> 
> const preview: Preview = {
>   globalTypes: {
>     locale: {
>       description: 'Internationalization locale',
>       defaultValue: 'en',
>       toolbar: {
>         icon: 'globe',
>         items: [
>           {
>             value: 'en',
>             right: '🇺🇸',
> -            left: '＄'
> +            icon: 'facehappy'
>             title: 'English'
>           },
>           { value: 'fr', right: '🇫🇷', title: 'Français' },
>           { value: 'es', right: '🇪🇸', title: 'Español' },
>           { value: 'zh', right: '🇨🇳', title: '中文' },
>           { value: 'kr', right: '🇰🇷', title: '한국어' },
>         ],
>       },
>     },
>   },
> };
> 
> export default preview;
> ```
> 
> The icon property only supports a limited set of icons, which are defined here:
> https://storybook.js.org/docs/8.0/faq#what-icons-are-available-for-my-toolbar-or-my-addon

Reviewer: @valentinpalkovic, when you have a moment, can you adjust this to:

````suggestion
The `TooltipLinkList` UI component used to customize the Storybook toolbar has been updated to use the `icon` property instead of the `left` property to position its content. If you've enabled this property in your `globalTypes` configuration, addons, or any other place, you'll need to replace it with an `icon` property to mimic the same behavior. For example:

```diff
// .storybook/preview.js|ts
// Replace your-framework with the framework you are using (e.g., react, vue3)
import { Preview } from '@storybook/your-framework';

const preview: Preview = {
  globalTypes: {
    locale: {
      description: 'Internationalization locale',
      defaultValue: 'en',
      toolbar: {
        icon: 'globe',
        items: [
          {
            value: 'en',
            right: '🇺🇸',
-            left: '＄'
+            icon: 'facehappy'
            title: 'English'
          },
          { value: 'fr', right: '🇫🇷', title: 'Français' },
          { value: 'es', right: '🇪🇸', title: 'Español' },
          { value: 'zh', right: '🇨🇳', title: '中文' },
          { value: 'kr', right: '🇰🇷', title: '한국어' },
        ],
      },
    },
  },
};

export default preview;
```
To learn more about the available icons and their names, see the [Storybook documentation](https://storybook.js.org/docs/8.0/faq#what-icons-are-available-for-my-toolbar-or-my-addon).
````
The rationale for this is that this change will affect Storybook's customization via `globalTypes` and addons (ours and from the community), correct?
